src/Language/Source.php

```diff
@@ -28,7 +33,8 @@ public function __construct($body, $name = null)
         );
 
         $this->body = $body;
-        $this->length = mb_strlen($body, 'UTF-8');
+        $this->buf =  array_map('GraphQL\Utils::ord', preg_split('//u', $body, -1, PREG_SPLIT_NO_EMPTY));
```

Comment:
@aelnagger I started experimenting with separate utf8 stream, but we should avoid duplicating our efforts, so please let me know if you are working on other solution now? If not, I'll continue.

Also, couple notes on best practices (regardless of our final solution):

1. GraphQL queries shouldn't be big. We should use variables as much as possible instead of inline values (because variables are parsed on C-level by `json_encode`).

2. Ideally parsing should be done by C module, so the most important issue for performance is #108. When it is implemented you should be able to use PHP module for parsing (e.g. https://github.com/dosten/graphql-parser-php) and then just pass this array to validation/execution steps.